numba/core/ir.py

```diff
@@ -26,7 +26,7 @@ class Loc(object):
     """Source location
 
     """
-    _defmatcher = re.compile(r'def\s+(\w+)\(.*')
+    _defmatcher = re.compile(r'def\s+(\w+)\s*[\[\(]')
```

Comment:
`_raw_function_name` is the only function that uses this regex. If we only need the name, `r'def\s+(\w+)'` should be enough, no?

Reply:
you're right.